### ISSUE TITLE
Result difference between uniseg.GraphemeClusterCount and Android android.icu.text.BreakIterator

### ISSUE THREAD

**Issue by munpf** [rivo/uniseg#60]
Hello!

I found that in the following text, the results of the two libraries are different. I'm not sure which one is correct, could you help to confirm if the result is expected with uniseg?

Thank you!

---
**Text:**
```क्‍ष```
it's unicode codepoint is
```"\u0915\u094d\u200d\u0937"```

**uniseg.GraphemeClusterCount Result:**
```2```

**android.icu.text.BreakIterator Result:**
```1```

**Comment by shogo82148** [comment]
what version of uniseg and android.icu do you use?

**Comment by rivo** [comment]
So the codepoints you posted are assigned the following grapheme break properties according to [this table](https://www.unicode.org/Public/15.0.0/ucd/auxiliary/GraphemeBreakProperty.txt):

1. `\u0915` = Any
2. `\u094d` = Extend
3. `\u200d` = Zero-Width Joiner
4. `\u0937` = Any

The way I see it, the following [rules](https://unicode.org/reports/tr29/#Grapheme_Cluster_Boundary_Rules) apply:

- **GB9** "Do not break before extending characters or ZWJ.": This applies to the sequence (1) (2) (3).
- **GB999** "Otherwise, break everywhere.": This applies to the sequence (3) (4).

So basically, `\u0937` is its own character according to those rules and must be counted separately. Unless I'm missing something, these are two separate characters, i.e. `2`. Incidentally, in Chrome where I'm writing this comment, I can select each character individually, too. So at least Chrome seems to agree.

I have no experience with android.icu so I can't comment on its functionality.

**Comment by munpf** [comment]
> what version of uniseg and android.icu do you use?

uniseg version is 0.4.7
android.icu has no version, it is a builtin library, and is copied from com.ibm.icu.text

**Comment by munpf** [comment]
@shogo82148  I searched and found that Android may use the official java implementation of icu. I wonder if there is any difference between the implementation of uniseg and the standard icu?

- [icu4j](https://unicode-org.github.io/icu/userguide/icu4j/), and Android use RuleBasedBreakIterator to iterate the text.
<img width="1758" alt="image" src="https://github.com/user-attachments/assets/5e59227f-8060-400f-9961-7b605c56142f">

---
In addition, I found that Android seems consider multiple consecutive "\n" as one grapheme cluster, while uniseg will consider it as multiple
